src/main/scala/is/hail/expr/AST.scala

```diff
@@ -662,8 +662,16 @@ case class Apply(posn: Position, fn: String, args: Array[AST]) extends AST(posn,
 
   def toIR(agg: Option[String] = None): Option[IR] = {
     fn match {
-      case "merge" | "select" | "drop" | "annotate" | "index" =>
+      case "merge" | "select" | "drop" | "index" =>
         None
+      case "annotate" =>
```

Comment:
Why is this special? is there something wrong with doing primo conversion on it?

Reply:
The AST for `Apply("annotate", Seq(structA,structB))` doesn't place limitations on what structB is, since it just uses structA.typ.annotate(structB.typ). I don't want to handle the general case, so I'm only converting in the case that the annotate node looks like `Apply("annotate", Seq(structA, StructConstructor(...)))`, which is I believe enough to handle all of our current uses in the Python.

Reply:
will something like this work?

```scala
def maybeFindIRFunction(fn: String, args: Seq[AST]): Option[IR] = for {
  irArgs <- anyFailAllFail(args.map(_.toIR(agg)))
  ir <- tryPrimOpConversion(args.map(_.`type`).zip(irArgs)).orElse(
    IRFunctionRegistry.lookupFunction(fn, args.map(_.`type`))
      .map { irf => irf(irArgs) })
} yield ir
```
and
```scala
fn match {
  // ...
  case "annotate" =>
    if (!args(1).isInstanceOf[StructConstructor])
      None
    else
      maybeFindIRFunction(fn, args)
  case _ =>
    maybeFindIRFunction(fn, args)
```

?

Reply:
yes. I'll change.